mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,222 @@
+    sfreq : int | float
+        The sampling frequency used to convert times into samples.
+    feature_names : array, shape (n_features,) | None
+        Names for input features to the model. If None, feature names will
+        be auto-generated from the shape of input data after running `fit`.
+    model : instance of sklearn model | string | None
+        The model used in fitting inputs and outputs. This can be any
+        sklearn-style model that contains a fit and predict method. Currently
+        the only supported string is 'ridge', which uses the default
+        :mod:`sklearn` ridge model. This is also what `None` defaults to.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features, n_delays)
+        The coefficients from the model fit, reshaped for easy visualization.
+        If you want the raw (2d) coefficients, use `mne.decoding.get_coefs`
+        on `self.model`.
+    mask_pred : array, dtype bool, shape (n_times,)
+        A mask with True values corresponding to datapoints that were used
+        in generating model predictions. Creating time delays necessitates that
+        datapoints at the beginning / end of the input are not used.
+
+    References
+    ----------
+    .. [1] Theunissen, F. E. et al. Estimating spatio-temporal receptive
+           fields of auditory and visual neurons from their responses to
+           natural stimuli. Network 12, 289-316 (2001).
+
+    .. [2] Willmore, B. & Smyth, D. Methods for first-order kernel
+           estimation: simple-cell receptive fields from responses to
+           natural scenes. Network 14, 553-77 (2003).
+
+    .. [3] Crosse, M. J., Di Liberto, G. M., Bednar, A. & Lalor, E. C. (2016).
+           The Multivariate Temporal Response Function (mTRF) Toolbox:
+           A MATLAB Toolbox for Relating Neural Signals to Continuous Stimuli.
+           Frontiers in Human Neuroscience 10, 604.
+           doi:10.3389/fnhum.2016.00604
+    """
+
+    def __init__(self, tmin, tmax, sfreq,
+                 feature_names=None, model=None):  # noqa: D102
+        from sklearn.linear_model import Ridge
+        model_dict = dict(ridge=Ridge())
+
+        self.feature_names = feature_names
+        self.sfreq = sfreq
+
+        self.delays = _times_to_delays(tmin, tmax, sfreq)
+        self.tmin = tmin
+        self.tmax = tmax
+
+        if isinstance(model, str):
+            if model not in list(model_dict.keys()):
+                raise ValueError('If string, model must be one'
+                                 ' of %s' % model_dict.keys())
+            model = model_dict[model]
+        self.model = Ridge() if model is None else model
+        _check_estimator(self.model)
+
+    def __repr__(self):  # noqa: D105
+        str_features = self.feature_names
+        s = "features : [%s, %s]" % (str_features[0], str_features[-1])
+        s += ", delays : [%f, %f]" % (self.delays[0], self.delays[-1])
+        s += ", model : %s" % str(self.model)
```

Comment:
good idea. think I should do this with a specific attribute (e.g. `fit_ == True`, or just check whether `delays_` exists since this will now be done at fitting time?)